app/screens/channel_info/channel_info.js

```diff
@@ -402,6 +459,14 @@ export default class ChannelInfo extends PureComponent {
         return isDirectMessage || isGroupMessage;
     };
 
+    renderUnarchiveChannel = (channelIsArchived) => {
```

Comment:
I need to the correct logic to know if the user has permission to unarchive channel